### PR TITLE
Merge 1.3.6

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

File: Library/.rubocop.yml [hunk 1]
```diff
@@ -211,6 +214,7 @@ Naming/HeredocDelimiterNaming:
 # we output how to use interpolated strings too often
 Lint/InterpolationCheck:
   Enabled: false
+<<<<<<< HEAD
```
Comment (maxim-belkin) [inline Library/.rubocop.yml]: Em... :)

File: Library/.rubocop.yml [hunk 2]
```diff
@@ -223,3 +227,5 @@ Layout/EmptyLineAfterMagicComment:
   Enabled: true
   Exclude:
     - '**/bin/**/*'
+=======
+>>>>>>> 1.3.6
```
Comment (maxim-belkin) [inline Library/.rubocop.yml]: Hm... :)

File: Library/Homebrew/diagnostic.rb [hunk 3]
```diff
@@ -811,6 +813,18 @@ def check_coretap_git_origin
...
           EOS
         end
+
+        return if ENV["CI"] || ENV["JENKINS_HOME"]
+
+        branch = coretap_path.git_branch
+        return if branch.nil? || branch =~ /master/
+
+        <<-EOS.undent
+          Homebrew/homebrew-core is not on the master branch
```
Comment (maxim-belkin) [inline Library/Homebrew/diagnostic.rb:823]: I think `Homebrew` has to be replaced with a variable that == `Linuxbrew` on Linux
@sjackman

File: Library/Homebrew/requirements/linux_requirement.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class LinuxRequirement < Requirement
+  fatal true
+
+  satisfy(build_env: false) { OS.linux? }
```
Comment (maxim-belkin) [inline Library/Homebrew/requirements/linux_requirement.rb:4]: @sjackman: do we want to add a check for Linux kernel version?

**Comment by iMichka** [comment]
```
failed: brew style

        ==> Installing or updating 'rubocop' gem
Building native extensions.  This could take a while...
Successfully installed rainbow-2.2.2
Successfully installed ast-2.3.0
Successfully installed parser-2.4.0.0
Successfully installed powerpack-0.1.1
Successfully installed ruby-progressbar-1.9.0
Successfully installed unicode-display_width-1.3.0
Successfully installed parallel-1.12.0
Successfully installed rubocop-0.50.0
8 gems installed
{"XDG_CACHE_HOME"=>"/home/linuxbrew/.cache/Homebrew/style"} rubocop _0.50.0_ --force-exclusion --parallel --except FormulaAudit,FormulaAuditStrict,NewFormulaAudit --config /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/.rubocop.yml /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew
(/home/linuxbrew/.linuxbrew/Homebrew/Library/.rubocop.yml): could not find expected ':' while scanning a simple key at line 217 column 1
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/psych.rb:379:in `parse'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/psych.rb:379:in `parse_stream'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/psych.rb:327:in `parse'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/psych.rb:293:in `safe_load'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/config_loader.rb:216:in `yaml_safe_load'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/config_loader.rb:199:in `load_yaml_configuration'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/config_loader.rb:34:in `load_file'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/config_loader.rb:99:in `block in base_configs'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/config_loader.rb:87:in `map'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/config_loader.rb:87:in `base_configs'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/config_loader_resolver.rb:21:in `resolve_inheritance'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/config_loader.rb:43:in `load_file'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/config_store.rb:23:in `options_config='
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/cli.rb:70:in `act_on_options'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/lib/rubocop/cli.rb:29:in `run'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/bin/rubocop:13:in `block in <top (required)>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
/home/linuxbrew/.gem/ruby/2.3.0/gems/rubocop-0.50.0/bin/rubocop:12:in `<top (required)>'
/home/linuxbrew/.gem/ruby/2.3.0/bin/rubocop:22:in `load'
/home/linuxbrew/.gem/ruby/2.3.0/bin/rubocop:22:in `<main>'
```

**Review by maxim-belkin — CHANGES_REQUESTED** [review]
I don't see any major problems (I noted minor ones in my comments) with this merge

File: Library/.rubocop.yml (hunk 1 above)
Comment (iMichka) [reply Library/.rubocop.yml]: Woups missed that one :D

**Comment by iMichka** [comment]
I fixed the merge <<< >>> stuff that I forgot in the one rubocop file. The merge was easy, though multiple files needed to be tweaked. I saw also 2-3 places were code converged between the mac/Linux version, which is a good thing.

For the changes in diagnostic.rb, this can be done in a separate commit on top of this merge, or even separate pull request if you prefer?

**Comment by iMichka** [comment]
We only have 3 style errors:
```
Offenses:

/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cmd/install.rb:330:3: W: Do not suppress exceptions.
  rescue FormulaUnavailableError
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/bottle.rb:64:1: C: Module has too many lines. [389/367]
module Homebrew ...
^^^^^^^^^^^^^^^
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/pull.rb:55:1: C: Module has too many lines. [387/367]
module Homebrew ...
^^^^^^^^^^^^^^^
```

File: Library/Homebrew/diagnostic.rb (hunk 3 above)
Comment (sjackman) [reply Library/Homebrew/diagnostic.rb:823]: I'm fine with this as is. We refer to this tap as `Homebrew/core` even when it's our Linuxbrew fork.

File: Library/Homebrew/requirements/linux_requirement.rb (hunk 4 above)
Comment (sjackman) [reply Library/Homebrew/requirements/linux_requirement.rb:4]: Currently the only formula that has a minimum Linux kernel version requirement is `glibc`. So I don't feel there's a need for this feature. If ≥ 3 formulae require  a minimum version of the Linux kernel, then yes.

**Comment by sjackman** [comment]
```
C: Module has too many lines. [389/367]
```
To address this failure, increase the maximum 367 to 389 in the Rubocop configuration.

**Comment by sjackman** [comment]
No `brew tests` failures!? That's a small miracle. ✨

File: Library/.rubocop.yml [hunk 5]
```diff
@@ -63,26 +63,26 @@ Metrics/AbcSize:
...
-  Max: 250
+  Max: 222
 
 Metrics/ModuleLength:
   CountComments: false
-  Exclude:
-    - '**/bin/**/*'
-    - '**/cmd/**/*'
-    - '**/lib/**/*'
+  Max: 367
```
Comment (sjackman) [inline Library/.rubocop.yml]: Increase this value to 389.

**Review by sjackman — CHANGES_REQUESTED** [review]
Thanks, Michka!

File: Library/.rubocop.yml (hunk 5 above)
Comment (iMichka) [reply Library/.rubocop.yml]: Done

**Review by maxim-belkin — APPROVED** [review]

**Comment by sjackman** [comment]
I forget the best way to address this failure. Try perhaps…
```ruby
  begin
    glibc = Formula["glibc"]
    interpreter = glibc.installed? ? glibc.lib/"ld-linux-x86-64.so.2" : sys_interpreter
    mkdir_p HOMEBREW_PREFIX/"lib"
    FileUtils.ln_sf interpreter, ld_so
  rescue FormulaUnavailableError
  end
```
or if necessary
```ruby
rescue FormulaUnavailableError # rubocop:disable Lint/RescueException
```

**Review by sjackman — APPROVED** [review]

**Comment by sjackman** [comment]
I'm happy with this merge once CI is 💚

**Comment by iMichka** [comment]
Merged. Let's wait 1 day or two before pushing the tag :)

**Comment by sjackman** [comment]
Thanks, Michka!

**Pull request merged**